### ISSUE TITLE
New Forms have invalid creation date

### ISSUE THREAD

**Issue by codesman** [wpninjas/ninja-forms#2061]
The default "Contact Me" form has a creation date, but any form that I've created has only zeroes for the date value:

![screen shot 2016-11-04 at 10 48 09 am](https://cloud.githubusercontent.com/assets/1653589/20016358/5faeb0b2-a27c-11e6-92e0-b4e639ca3aa1.png)

**Comment by isleshocky77** [comment]
I came to report this same issue. The upgrade from 2 to 3 actually killed all my previous form's creation dates as well as all new forms having "0000-00-00 00:00:00" "Date Created" values.

**Comment by isleshocky77** [comment]
h2. Problem
It looks like this is happening due to trying to insert a unix epoch in a MySQL `timestamp` field. "

> TIMESTAMP has a range of '1970-01-01 00:00:01' UTC to '2038-01-19 03:14:07' UTC

 - [Source](https://dev.mysql.com/doc/refman/5.5/en/datetime.html)

## Table Definition
```php
# /plugins/ninja-forms/includes/Database/Migrations/Fields.php:13
    public function run()
    {
        $query = "CREATE TABLE IF NOT EXISTS $this->table_name (
            `id` int NOT NULL AUTO_INCREMENT,
            `label` longtext,
            `key` longtext,
            `type` longtext,
            `parent_id` int NOT NULL,
            `created_at` TIMESTAMP,
            `updated_at` DATETIME,
            UNIQUE KEY (`id`)
        ) $this->charset_collate;";
```
## Model object creation
```php
# /plugins/ninja-forms/includes/Abstracts/Model.php:473
    public function save()
    {
        // If the ID is not set, assign an ID
        if( ! $this->_id ){

            $data = array( 'created_at' => time() );

            if( $this->_parent_id ){
                $data['parent_id'] = $this->_parent_id;
            }

            // Create a new row in the database
            $result = $this->_db->insert(
                $this->_table_name,
                $data
            );

            // Assign the New ID
            $this->_id = $this->_db->insert_id;
        } else {
```

h3. Solutions

* Update the table definition to have a default value of `CURRENT_TIMESTAMP` and create a migration
* Update object initialization to create a properly formatted date (i.e. 2017-04-25 22:15:18) (e.g. `date('Y-m-d H:i:s')`

I'm not the author of this code or the project so I don't know which would be the preferred for a pull request.

**Comment by isleshocky77** [comment]
@kstover Thoughts on the two offered solutions above? Would love to see this fixed.

**Comment by kstover** [comment]
@isleshocky77 I think that the best solution would be to update how our code handles inserting the creation date, not the SQL column type.

If you can submit a PR, I'll get it merged.

**Comment by isleshocky77** [comment]
@kstover Submitted. Seems to work properly in my quick testing.

**Comment by kstover** [comment]
Thanks! I'll take a look and see about getting it merged in for version 3.1.5!
